src/tree.rs

```diff
@@ -660,6 +661,32 @@ impl<T> Node<T> {
     }
 }
 
+impl<T> Node<T> {
+    /// Iterates over the tree and calls the given visitor function
+    /// with fully resolved path and its value.
+    pub fn for_each<V: FnMut(String, T) -> bool >(self, mut visitor: V)  {
+        let mut queue = VecDeque::new();
+        queue.push_back(self);
+        while let Some(mut node) = queue.pop_front() {
+            if node.children.is_empty() {
+                denormalize_params(&mut node.prefix, &node.remapping);
+                let path = String::from_utf8(node.prefix.into_unescaped()).unwrap();
+                let value = node.value.take().map(UnsafeCell::into_inner);
+                if !visitor(path, value.unwrap()) {
+                    return;
+                }
+            } else {
+                for mut child in node.children {
+                    let mut prefix = node.prefix.clone();
+                    prefix.append(&child.prefix);
+                    child.prefix = prefix;
```

Comment:
Does this mutate the child prefix permanently? If so it would break the prefix search. You can keep a queue of `(String, Node)` instead of mutating here.